SalesforceCommerceCloud/pwa-kit
# [App Extensibility] Get the correct configured extensions at runtime (@W-17109413@)

There are 2 concepts of extensions: installed and configured.  There can possibly be different configs for different environments. So we wanted to make sure that we get the correct _configured_ extensions, and the way to do that is to wait until runtime.

This PR delays as many `getConfig` call as what makes sense, in the context of App Extensibility.

- [x] Fix the tests and types

## How to test drive the PR

1. `npm ci` at the monorepo root
2. `npm start` at the typescript-minimal project
3. Verify that the project renders with a blue border around the app. That's the result of the sample extension enabled.
4. Now edit the package.json to disable the sample extension
5. Wait for the dev server to rebuild. No need to restart the server because the configured extensions are now parsed during runtime.
6. Reload your browser and verify that the homepage no longer has the blue border.
